File: ChatApp/Hubs/CommentRealTime.cs
using ChatApp.Models.Dto;
using Microsoft.AspNet.SignalR;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ChatApp.Hubs
{
    public partial class ChatHub : Hub
    {
        /// <summary>
        /// Tạo bài đăng
        /// Created By: NBDuong 22.05.2020
        /// </summary>
        /// <param name="GroupName"></param>
        /// <param name="postDto"></param>
        /// <param name="notiDto"></param>
        public void CreatePostNew(string GroupName,PostDto postDto ,NotifiDto notiDto)
        {
            Clients.All.CreatePost(GroupName, postDto,notiDto);
        }

        /// <summary>
        /// Tạo comment cấp 1
        /// Created By: NBDuong 24.05.2020
        /// </summary>
        /// <param name="myuserName"></param>
        /// <param name="groupName"></param>
        /// <param name="postId"></param>
        /// <param name="commentDto"></param>
        /// <param name="notiDto"></param>
        public void CreateCommentNew(string myuserName,string groupName,int postId, CommentDto commentDto,NotifiDto notiDto)
        {
            Clients.All.CreateComment(groupName, postId, commentDto);
            if (myuserName != commentDto.UserName)
            {
                List<string> listId = GetListConnectIdByUserName("No value", commentDto.UserName);
                Clients.Clients(listId).UpdateNotiRealtime(notiDto);
            }
        }
        
        /// <summary>
        /// Tạo comment cấp 2 
        /// Created By: NBDuong 26.05.2020
        /// </summary>
        /// <param name="myuserName"></param>
        /// <param name="groupName"></param>
        /// <param name="commentId"></param>
        /// <param name="subCommentDto"></param>
        /// <param name="notiDto"></param>
        public void CreateSubCommentNew(string myuserName,string groupName, int commentId, SubCommentDto subCommentDto,NotifiDto notiDto)
        {
            Clients.All.CreateSubComment(groupName, commentId, subCommentDto);
            if (myuserName != subCommentDto.UserName)
            {
                List<string> listId = GetListConnectIdByUserName("No value", subCommentDto.UserName);
                Clients.Clients(listId).UpdateNotiRealtime(notiDto);
            }
        }

        /// <summary>
        /// Xóa bài đăng
        /// Created by: NBDuong 27.05.2020
        /// </summary>
        /// <param name="groupName"></param>
        /// <param name="postId"></param>
        public void DeletePost(string groupName, int postId)
        {
            Clients.All.DeletePost(groupName, postId);
        }

        /// <summary>
        /// Xóa comment cấp 1
        /// Created By: NBDuong 27.05.2020
        /// </summary>
        /// <param name="groupName"></param>
        /// <param name="commentId"></param>
        public void DeleteComment(string groupName, int commentId)
        {
            Clients.All.DeleteComment(groupName, commentId);
        }

        /// <summary>
        /// Xóa comment cấp 2
        /// Created By: NBDuong 27.05.2020
        /// </summary>
        /// <param name="groupName"></param>
        /// <param name="subcommentId"></param>
        public void DeleteSubComment(string groupName, int subcommentId)
        {
            Clients.All.DeleteSubComment(groupName, subcommentId);
        }

        /// <summary>
        /// Sửa bài post
        /// Created By: NBDuong 28.05.2020
        /// </summary>
        /// <param name="groupName"></param>
        /// <param name="postDto"></param>
        public void EditPost(string groupName, PostDto postDto)
        {
            Clients.All.EditPost(groupName, postDto);
        }

        /// <summary>
        /// Sửa comment cấp 1
        /// Created by: NBDuong 28.05.2020
        /// </summary>
        /// <param name="groupName"></param>
        /// <param name="commentDto"></param>
        public void EditComment(string groupName, CommentDto commentDto)
        {
            Clients.All.EditComment(groupName, commentDto);
        }

        /// <summary>
        /// Sửa comment cấp 2
        /// Created By: NBDuong 28.05.2020
        /// </summary>
        /// <param name="groupName"></param>
        /// <param name="subcommentDto"></param>
        public void EditSubComment(string groupName, SubCommentDto subcommentDto)
        {
            Clients.All.EditSubComment(groupName, subcommentDto);
        }

        /// <summary>
        /// Like bài post
        /// Created By: NBDuong 29.05.2020
        /// </summary>
        /// <param name="myuserName"></param>
        /// <param name="likeDto"></param>
        /// <param name="postId"></param>
        /// <param name="groupName"></param>
        public void SaveLikePost(string myuserName, LikeDto likeDto, int postId, string groupName)
        {
            Clients.All.SaveLikePost(likeDto.Check, postId, groupName);
            if (myuserName != likeDto.UserName)
            {
                List<string> listId = GetListConnectIdByUserName("No value", likeDto.UserName);
                Clients.Clients(listId).UpdateNotiRealtime();
            }
        }

        /// <summary>
        /// Like comment cấp 1
        /// Created By: NBDuong 29.05.2020
        /// </summary>
        /// <param name="myuserName"></param>
        /// <param name="likeDto"></param>
        /// <param name="commentId"></param>
        /// <param name="groupName"></param>
        public void SaveLikeComment(string myuserName, LikeDto likeDto, int commentId, string groupName)
        {
            Clients.All.SaveLikeComment(likeDto.Check, commentId, groupName);
            if (myuserName != likeDto.UserName)
            {
                List<string> listId = GetListConnectIdByUserName("No value", likeDto.UserName);
                Clients.Clients(listId).UpdateNotiRealtime();
            }
        }

        /// <summary>
        /// Like comment cấp 2
        /// Created By: NBDuong 29.05.2020
        /// </summary>
        /// <param name="myuserName"></param>
        /// <param name="likeDto"></param>
        /// <param name="subCommentId"></param>
        /// <param name="groupName"></param>
        public void SaveLikeSubComment(string myuserName, LikeDto likeDto, int subCommentId, string groupName)
        {
            Clients.All.SaveLikeSubComment(likeDto.Check, subCommentId, groupName);
            if (myuserName != likeDto.UserName)
            {
                List<string> listId = GetListConnectIdByUserName("No value", likeDto.UserName);
                Clients.Clients(listId).UpdateNotiRealtime();
            }
        }

        /// <summary>
        /// Gửi lời mời kết bạn
        /// Created By: NBDuong 30.05.2020
        /// </summary>
        /// <param name="username"></param>
        public void SendRequestAddFriend(string username)
        {
            List<string> listId = GetListConnectIdByUserName("No value", username);
            Clients.Clients(listId).UpdateNotiRequestAddFriend();
        }
    }
}

File: ChatApp/Assets/Script/CommentRealTime.js

$(function () {
    hub.client.createPost = function (GroupName, postDto) {
        name = $(".ToolFb .ToolLeft h2:eq(0)").text().trim();
        if (postDto.UserName !== MyUserName) {
            countNoti = parseInt($(".badge").text()) + 1;
            $(".badge").text(countNoti).show();
        }
        if (name === GroupName) {
            $(".post-clone .avatar-post").find('img').attr('src', postDto.avatar);
            $(".post-clone").find('.name-user a').text(postDto.NameOfUser).attr("href",postDto.UrlProfile);
            $(".post-clone").find('.time-post').text(postDto.TimePost);
            $(".post-clone").find('.content-post').eq(1).find('p').html(postDto.PostText);
            $(".post-clone").find('.countLike_post').text(postDto.LikeNumber);
            $(".post-clone .post").attr('id', postDto.PostId);
            var demo = $(".post-clone").html();
            $(".post-clone .post").attr('id', 0);
            $(".post-append").prepend(demo);
            if (postDto.UserName !== MyUserName) {
                $(".post-append .post").eq(0).find('.edit-post').remove();
                $(".post-append .post").eq(0).find('.delete-post').remove();
            }
        }
    };

    hub.client.createComment = function (groupName, postId, commentDto) {
        name = $(".ToolFb .ToolLeft h2:eq(0)").text().trim();
        Post = $(".post[id=" + postId + "]");
        if (name === groupName) {
            $('.post-clone .comment-level .img-comment img').attr('src', commentDto.Avatar);
            $('.post-clone .name-comment').text(commentDto.NameOfUser).attr("href", commentDto.UrlProfile);
            $('.post-clone .name-comment').next().text(commentDto.Text);
            $('.post-clone .comment-level').attr('id', commentDto.CommentId);
            var demo = $('.post-clone .comment-level').parent().html();
            $('.post-clone .comment-level').attr('id',0);
            Post.find('.comment-post-wrapper').append(demo);
            if (commentDto.UserNameComment !== MyUserName) {
                Post.find('.comment-post-wrapper .comment-level').last().find('.comment-content_setting').remove();
            }
        }
    };

    hub.client.updateNotiRealtime = function () {
            countNoti = parseInt($(".badge").text()) + 1;
            $(".badge").text(countNoti).show();
    };

    hub.client.createSubComment = function (groupName, commentId, subCommentDto) {
        name = $(".ToolFb .ToolLeft h2:eq(0)").text().trim();
        Comment = $(".comment-level[id=" + commentId + "]");
        if (name === groupName) {
            $('.post-clone .comment-level2 img:eq(0)').attr('src', subCommentDto.Avatar);
            $('.post-clone .comment-level2').find('.name-reply').eq(0).text(subCommentDto.NameOfUser).attr("href", subCommentDto.UrlProfile);;
            $('.post-clone .comment-level2').find('.name-reply').eq(0).next().text(subCommentDto.Text);
            $('.post-clone .comment-level2').attr('id', subCommentDto.SubCommentId);
            var demo2 = $('.post-clone .comment-level2').parent().html();
            $('.post-clone .comment-level2').attr('id',0);
            Comment.find('.comment-wrapper2').append(demo2);
            if (subCommentDto.UserNameComment !== MyUserName) {
                Comment.find('.comment-wrapper2 .comment-level2').last().find('.reply-content_setting').remove();
            }
        }
    };

    hub.client.deletePost = function (groupName, postId) {
        name = $(".ToolFb .ToolLeft h2:eq(0)").text().trim();
        Post = $(".post[id=" + postId + "]");
        if (name === groupName) {
            Post.remove();
        }
    };

    hub.client.deleteComment = function (groupName, commentId) {
        name = $(".ToolFb .ToolLeft h2:eq(0)").text().trim();
        Comment = $(".comment-level[id=" + commentId + "]");
        if (name === groupName) {
            Comment.remove();
        }
    };

    hub.client.deleteSubComment = function (groupName, subcommentId) {
        name = $(".ToolFb .ToolLeft h2:eq(0)").text().trim();
        SubComment = $(".comment-level2[id=" + subcommentId + "]");
        if (name === groupName) {
            SubComment.remove();
        }
    };

    hub.client.editPost = function (groupName, postDto) {
        name = $(".ToolFb .ToolLeft h2:eq(0)").text().trim();
        Post = $(".post[id=" + postDto.PostId + "]");
        if (name === groupName) {
            Post.find(".content-post p:eq(0)").text(postDto.PostText);
        }
    };

    hub.client.editComment = function (groupName, commentDto) {
        name = $(".ToolFb .ToolLeft h2:eq(0)").text().trim();
        Comment = $(".comment-level[id=" + commentDto.CommentId + "]");
        if (name === groupName) {
            Comment.find(".comment-content span:eq(0)").text(commentDto.Text);
        }
    };

    hub.client.editSubComment = function (groupName, subcommentDto) {
        name = $(".ToolFb .ToolLeft h2:eq(0)").text().trim();
        SubComment = $(".comment-level2[id=" + subcommentDto.SubCommentId + "]");
        if (name === groupName) {
            SubComment.find(".reply-content span:eq(0)").text(subcommentDto.Text);
        }
    };

    hub.client.saveLikePost = function (check, postId, groupName) {
        name = $(".ToolFb .ToolLeft h2:eq(0)").text().trim();
        Post = $(".post[id=" + postId + "]");
        if (name === groupName) {
            if (check) {
                countLike = parseInt(Post.find(".countLike_post").text()) + 1;
                Post.find(".countLike_post").text(countLike);
            } else {
                countLike = parseInt(Post.find(".countLike_post").text()) - 1;
                Post.find(".countLike_post").text(countLike);
            }
        }
    };

    hub.client.saveLikeComment = function (check, commentId, groupName) {
        name = $(".ToolFb .ToolLeft h2:eq(0)").text().trim();
        Comment = $(".comment-level[id=" + commentId + "]");
        if (name === groupName) {
            if (check) {
                countLike = parseInt(Comment.find(".countLike_comment").text()) + 1;
                Comment.find(".countLike_comment").text(countLike);
            } else {
                countLike = parseInt(Comment.find(".countLike_comment").text()) - 1;
                Comment.find(".countLike_comment").text(countLike);
            }
        }
    };

    hub.client.saveLikeSubComment = function (check, subcommentId, groupName) {
        name = $(".ToolFb .ToolLeft h2:eq(0)").text().trim();
        SubComment = $(".comment-level2[id=" + subcommentId + "]");
        if (name === groupName) {
            if (check) {
                countLike = parseInt(SubComment.find(".countLike_reply").text()) + 1;
                SubComment.find(".countLike_reply").text(countLike);
            } else {
                countLike = parseInt(SubComment.find(".countLike_reply").text()) - 1;
                SubComment.find(".countLike_reply").text(countLike);
            }
        }
    };
});